lib/base-compiler.ts

```diff
@@ -3350,6 +3350,11 @@ but nothing was dumped. Possible causes are:
                       filters,
                   )
                 : (async () => {
+                      if (result.validatorTool && result.code === 0) {
```

Comment:
I spent time looking around, there are many things that rely on `result.asm`/`result.asmSize` and this spot

I figured a dedicated `validatorTool` was best incase the `asm/asmSize` is changed in the future. Because we don't set `asm`/`asmSize` I also need to put this first